python_ta/checkers/invalid_name_checker.py

```diff
@@ -0,0 +1,525 @@
+NAME_CHECK = {
+    "module": _check_module_name,
+    "const": _check_const_name,
+    "class": _check_class_name,
+    "function": _check_function_and_variable_name,
+    "method": _check_method_and_attr_name,
+    "attr": _check_method_and_attr_name,
+    "argument": _check_argument_name,
+    "variable": _check_function_and_variable_name,
+    "class_attribute": _check_method_and_attr_name,
+    "class_const": _check_const_name,
+    "inlinevar": _check_inlinevar_name,
+    "typevar": _check_typevar_name,
+    "typealias": _check_typealias_name,
+}
+
+
+class InvalidNameChecker(BaseChecker):
+    """A checker class to report on names that don't conform to PEP8 naming conventions.
+
+    For the PEP8 naming conventions, see https://peps.python.org/pep-0008/#naming-conventions.
+    """
+
+    name = "pep8_name_violation"
+    # Template of displayed message will include the following:
+    #   1. Describe the variable name type (e.g. `module`, `const`, etc.) and the name itself.
+    #   2. Explanation of how the name violates the naming convention and suggest a
+    #   correction(s), if appropriate.
+    msgs = {
+        "C9103": (
+            'Found %s name "%s" in violation of PEP8 naming conventions. %s',
+            "pep8-name-violation",
+            "Used when the name doesn't conform to naming conventions specified "
+            "in the PEP8 style guide.",
+        ),
+    }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @only_required_for_messages("pep8-name-violation")
+    def visit_module(self, node: nodes.Module) -> None:
+        """Visit a Module node to check for any name violations.
+
+        Snippets taken from pylint.checkers.base.name_checker.checker."""
+        self._check_name("module", node.name.split(".")[-1], node)
+
+    @only_required_for_messages("pep8-name-violation")
+    def visit_classdef(self, node: nodes.ClassDef):
+        """Visit a Class node to check for any name violations.
+
+        Taken from pylint.checkers.base.name_checker.checker."""
+        self._check_name("class", node.name, node)
+        # Check any instance attributes that were first defined in this class.
+        for attr, anodes in node.instance_attrs.items():
+            if not any(node.instance_attr_ancestors(attr)):
+                self._check_name("attr", attr, anodes[0])
+
+    @only_required_for_messages("pep8-name-violation")
+    def visit_functiondef(self, node: nodes.FunctionDef) -> None:
+        """Visit a FunctionDef node to check for any name violations.
+
+        Snippets taken from pylint.checkers.base.name_checker.checker."""
+        if node.is_method():
+            if utils.overrides_a_method(node.parent.frame(future=True), node.name):
+                return
+
+        self._check_name(
+            _determine_function_name_type(node, config=self.linter.config), node.name, node
+        )
+        # Check argument names
```

Comment:
Instead of putting this here, figure out what `visit_` method gets called on each parameter name, and do the check there. (`_recursive_check_names` isn't a very useful helper method, and should be removed)

Reply:
`visit_assignname` is called and we can use the existing variable `assign_type` to check if it was assigned in an `Arguments` node.